code/supervised/Logistic_Regression/logistic_ex1.py

```diff
@@ -0,0 +1,52 @@
+from sklearn.linear_model import LogisticRegression
+import numpy as np
+import random
+
+#defines the classification for the training data.
+def true_classifier(i):
+    if i >= 700:
+        return 1
+    return 0
+
+#Generate a random dataset which includes random scores from 0 to 1000.
+x = np.array([ random.randint(0,1000) for i in range(0,1000) ])
+
+#The model will expect a 2D array, so we must reshape
+#For the model, the 2D array must have rows equal to the number of samples,
+#and columns equal to the number of features.
+#For this example, we have 1000 samples and 1 feature.
+x = x.reshape((-1, 1))
+
+#For each point, y is a pass/fail for the grade. The simple threshold is arbitrary,
+#and can be changed as you would like. Classes are 1 for success and 0 for failure
+y = [ true_classifier(x[i][0]) for i in range(0,1000) ]
+
+
+#Again, numpy wants a 2d array, so we convert.
+y = np.array(y)
+y = y.reshape(-1,1)
+
+#Our goal will be to train a logistic regression model to do pass/fail to the same threshold.
+model = LogisticRegression()
+
+#The fit method actually fits the model to our training data
+model = model.fit(x,y)
+
+#Create 100 random samples to try against our model as test data
+samples = [random.randint(0,1000) for i in range(0,100)]
+#Once again, we need a 2d Numpy array
+samples = np.array(samples)
+samples = samples.reshape(-1, 1)
+
+#Now we use our model against the samples.  output is the probability, and _class is the class.
+_class = model.predict(samples)
+proba = model.predict_proba(samples)
+
+num_accurate = 0
+
+#Finally, output the results, formatted for nicer viewing.
```

Comment:
Explain a little on what the output means,  
for example, 
[673]: Class 0, probability [0.59449694 0.40550306]
explain the two probability numbers and the [673]: 


also i would put a new line before you print out 100 out of 100 correct, just looks a little nicer.

Reply:
Left a comment with the explanation and an example, and added a new line